src/utils/source.js

```diff
@@ -313,3 +313,11 @@ export function isLoaded(source: SourceRecord) {
 export function isLoading(source: SourceRecord) {
   return source.get("loadedState") === "loading";
 }
+
+export function getExtensionType(url: string) {
```

Comment:
This would break if the URL had a "?", for instance.

Instead of this function, it would be best to factor out `SourceTree`'s `getIcon` method as well as `sourceTypes` o bject.  We should extract as much of that logic as we can and make it a utility method in `src/utils/source.js`

Reply:
Question, why do we use <Svg/>  sometimes over <img/> ? What's the preferred usage?

Reply:
<img> + CSS is more performant than inline svgs